File: SAAMSONSDISTRIBUTORS/Models/Delivery.cs
// Delivery.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SAAMSONSDISTRIBUTORS.Models
{
    public class Delivery
    {
        [Key] 
        public int Id { get; set; }

        [Required] 
        public int ProductId { get; set; }
        public Product? Product { get; set; }

        [Required] 
        public int ClientId { get; set; }

        [Required, MaxLength(128)]
        public string UserId { get; set; } = string.Empty;

        [MaxLength(50)]
        public string InvoiceNumber { get; set; } = string.Empty;

        [Required]
        public CartStatus Status { get; set; } = CartStatus.Pending; // Pending, Delivery, In Progress

        [Required] public int Quantity { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal UnitPriceAtTime { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal DiscountPerProduct { get; set; } = 0m;

        [Column(TypeName = "decimal(5,2)")]
        public decimal PercentageDiscount { get; set; } = 0m;

        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalDiscount { get; set; } = 0m;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Total { get; set; } = 0m;

        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    }
}

File: SAAMSONSDISTRIBUTORSWEB/sam-sons-distributors/src/app/models/delivery.ts
import { Decimal, ISODateString } from './common';
import { Product } from './product';

export enum DeliveryStatus {
  Pending = 'Pending',
  OutForDelivery = 'OutForDelivery',
  Delivered = 'Delivered',
  Cancelled = 'Cancelled',
}

export interface Delivery {
  id: number;
  productId: number;
  product?: Product | null;

  clientId: number;
  userId: string;

  invoiceNumber: string;
  status: DeliveryStatus;

  quantity: number;

  unitPriceAtTime: Decimal;
  discountPerProduct: Decimal;
  percentageDiscount: Decimal;
  totalDiscount: Decimal;
  total: Decimal;

  createdDate: ISODateString;
}
